### PR TITLE
Cleanup array2.fs

### PR DESCRIPTION
**Pull request by forki**
* no need to box in isNull check
* small whitespace cleanup

### CODE REVIEW

File: src/fsharp/FSharp.Core/array2.fs [hunk 1]
```diff
@@ -95,20 +93,20 @@ namespace Microsoft.FSharp.Collections
...
-            let b1 = base1 array 
-            let b2 = base2 array 
+            let count1 = length1 array
+            let count2 = length2 array
+            let b1 = base1 array
+            let b2 = base2 array
             let f = OptimizedClosures.FSharpFunc<_,_,_,_>.Adapt(action)
-            for i = b1 to b1+count1 - 1 do 
-              for j = b2 to b2+count2 - 1 do 
+            for i = b1 to b1+count1 - 1 do
```
Comment (KevinRansom) [inline src/fsharp/FSharp.Core/array2.fs]: since it's cleanup spaces around the +   ?

Comment (forki) [reply src/fsharp/FSharp.Core/array2.fs]: done

File: src/fsharp/FSharp.Core/array2.fs [hunk 2]
```diff
@@ -95,20 +93,20 @@ namespace Microsoft.FSharp.Collections
...
+            let count1 = length1 array
+            let count2 = length2 array
+            let b1 = base1 array
+            let b2 = base2 array
             let f = OptimizedClosures.FSharpFunc<_,_,_,_>.Adapt(action)
-            for i = b1 to b1+count1 - 1 do 
-              for j = b2 to b2+count2 - 1 do 
+            for i = b1 to b1 + count1 - 1 do
+              for j = b2 to b2 + count2 - 1 do
                 f.Invoke(i, j, array.[i,j])
```
Comment (KevinRansom) [inline src/fsharp/FSharp.Core/array2.fs]: spaces on ,

File: src/fsharp/FSharp.Core/array2.fs [hunk 3]
```diff
@@ -95,20 +93,20 @@ namespace Microsoft.FSharp.Collections
...
             checkNonNull "array" array
-            let count1 = length1 array 
-            let count2 = length2 array 
-            let b1 = base1 array 
-            let b2 = base2 array 
+            let count1 = length1 array
+            let count2 = length2 array
+            let b1 = base1 array
+            let b2 = base2 array
             let f = OptimizedClosures.FSharpFunc<_,_,_,_>.Adapt(action)
```
Comment (KevinRansom) [inline src/fsharp/FSharp.Core/array2.fs]: spaces on ,

File: src/fsharp/FSharp.Core/array2.fs [hunk 4]
```diff
@@ -160,5 +158,3 @@ namespace Microsoft.FSharp.Collections
             for i = 0 to count1 - 1 do
                 for j = 0 to count2 - 1 do
                     target.[targetIndex1+i,targetIndex2+j] <- source.[sourceIndex1+i,sourceIndex2+j]
```
Comment (KevinRansom) [inline src/fsharp/FSharp.Core/array2.fs]: spaces on +

File: src/fsharp/FSharp.Core/array2.fs (hunk 3 above)
Comment (smoothdeveloper) [reply src/fsharp/FSharp.Core/array2.fs]: `IDictionary<int,string>` or `IDictionary<int, string>`, I'm not sure on this one, I believe in C# code, no spaces used.

https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting there doesn't seem to be recommendations for generic type parameter.

msdn doesn't use spaces in the formatting of the doc:

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.idictionary-2?view=netframework-4.7.2

Also looks odd if we enforce spaces after comma here but ban `IDictionary< int, string >`

Comment (KevinRansom) [reply src/fsharp/FSharp.Core/array2.fs]: @smoothdeveloper.

Cleanup PRs are very bike-sheddable, and I should know better.  However, no C# developer would write ````var lst = new List< String >()````.   And plenty would write ````var lst = new List<string, string>()````
Many places in our codebase we use the C# convention. So I am down with following it.

**Comment by KevinRansom** [comment]
@forki, Thanks for taking care of this.

**Pull request merged**